apps/depthwise_separable_conv/depthwise_separable_conv_generator.cpp

```diff
@@ -0,0 +1,300 @@
+                .split(pointwise_filter_dom, ro, ri, 4)
+                .reorder(ri, x, y, d, ro)
+                .unroll(ri);
+
+            // We're going to call in() on depthwise_convolved three
+            // times! The first will be to give it a compute_root
+            // wrapper to do the accumulation in registers. The second
+            // will be staging tiles of it into shared in the
+            // pointwise kernel. The third will be staging the loads
+            // from shared into registers. We write them in reverse order below:
+
+            // We can do 4-wide vectorized loads from shared memory if
+            // we unroll the reduction loop by a factor of four above
+            // and stage the loads from the depthwise_convolved
+            // output.
+            depthwise_convolved.in()
+                .in()
+                .in()
+                .compute_at(pointwise_convolved, x)
+                .bound_extent(d, 4)
+                .vectorize(d)
+                .unroll(x)
+                .unroll(y);
+
+            // Stage a tile depthwise_convolved into shared memory for
+            // each tile of pointwise_convolved.
+            depthwise_convolved.in()
+                .in()
+                .compute_at(output, d)
+                .tile({d, x, y}, {di, xi, yi}, {16, 2, 2}, TailStrategy::RoundUp)
+                .unroll(xi)
+                .unroll(yi)
+                .gpu_threads(di, x, y);
+
+            // The depthwise convolution kernel
+            depthwise_convolved.in()
+                .compute_root()
+                .tile({d, x, y}, {di, xi, yi}, {32, 4, 4})
+                .tile({di, xi, yi}, {dii, xii, yii}, {2, 2, 2})
+                .gpu_threads(di, xi, yi)
+                .fuse(y, b, b)
+                .gpu_blocks(d, x, b)
+                .unroll(xii)
+                .unroll(yii)
+                .unroll(dii);
+
+            depthwise_convolved
+                .compute_at(depthwise_convolved.in(), di)
+                .unroll(x)
+                .unroll(y)
+                .unroll(d)
+                .update()
+                .reorder(d, x, y, depthwise_filter_dom[1], depthwise_filter_dom[2], depthwise_filter_dom[0])
+                .unroll(x)
+                .unroll(y)
+                .unroll(d);
+        } else {
+            // CPU schedule
+
+            // 0.146ms on an Intel i9-9960X using 16 threads pinned to 3.0 GHz,
+            // which is only about 18% of peak flops.
+
+            int tile_w = 1;
+            int tile_h = 1;
+            int tile_d = 1;
+            const int vec = natural_vector_size<float>();
+
+            // Figure out how many registers we have in the register
+            // file on this target.
+            int num_regs = 16;
+            if (get_target().has_feature(Target::AVX512_Skylake) ||
+                (get_target().arch == Target::ARM &&
+                 get_target().bits == 64)) {
+                num_regs = 32;
```

Comment:
Maybe this should be a helper function of Target, similar to natural_vector_size?